index.js

```diff
@@ -161,16 +163,66 @@ async function handleSwitchToBranchCommand(sg, sortedBranches, combinedBranch) {
   process.exit(0);
 }
 
+/**
+ * @param {YamlCfg} ymlConfig
+ * @param {string} path
+ */
+function getConfigOption(ymlConfig, path) {
+  const parts = path.split(/\./g);
+  let ptr = ymlConfig;
+  while (ptr && parts.length) {
+    const part = parts.shift();
+    // cater for both kebab case and camel cased variants of key, just for developer convenience.
+    ptr = part in ptr ? ptr[part] : ptr[camelCase(part)];
+  }
+  return ptr;
+}
+
 async function main() {
+  const sg = simpleGit();
+  if (!(await sg.checkIsRepo())) {
+    console.log('Not a git repo'.red);
+    process.exit(1);
+  }
+
+  // try to create or init the config first so we can read values from it
+  if (process.argv.includes('--init')) {
+    if (fs.existsSync(await getConfigPath(sg))) {
+      console.log('.pr-train.yml already exists');
+      process.exit(1);
+    }
+    const root = path.dirname(require.main.filename);
+    const cfgTpl = fs.readFileSync(`${root}/cfg_template.yml`);
+    fs.writeFileSync(await getConfigPath(sg), cfgTpl);
+    console.log(`Created a ".pr-train.yml" file. Please make sure it's gitignored.`);
+    process.exit(0);
+  }
+
+  let ymlConfig;
+  try {
+    ymlConfig = await loadConfig(sg);
+  } catch (e) {
+    if (e instanceof yaml.YAMLException) {
+      console.log('There seems to be an error in `.pr-train.yml`.');
+      console.log(e.message);
+      process.exit(1);
+    }
+    console.log('`.pr-train.yml` file not found. Please run `git pr-train --init` to create one.'.red);
+    process.exit(1);
+  }
```

Comment:
By moving this block here, I was able to use the values read from the config to inform the command line parameters added or their default values.